Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -1419,7 +1419,8 @@ MediaTime MediaPlayerPrivateGStreamer::playbackPosition() const
         return m_cachedPosition;
     }
 
-    GstClockTime gstreamerPosition = gstreamerPositionFromSinks();
+    // We can't trust sinks position when pipeline is flushed (e.g. after MSE samples removal).
+    GstClockTime gstreamerPosition = isPipelineSeeking() ? GST_CLOCK_TIME_NONE : gstreamerPositionFromSinks();
```

Comment:
`isPipelineSeeking()` is a misleading name. It should be called instead something like `isPipelineWaitingPreroll()`. Also I'm seeing the condition inside of it contains `pending == GST_STATE_PAUSED`, shouldn't it be `pending >= GST_STATE_PAUSED`? I guess that usually works because we have a bunch of code for pausing the pipeline during seeks?

Reply:
Makes sense, thanks!